src/ppx/application.cpp

```diff
@@ -998,6 +998,21 @@ void Application::DispatchConfig()
         mSettings.window.title = mSettings.appName;
     }
 
+    // ImGui doesn't support prebuilt command buffer.
+    if (mSettings.grfx.enablePreRecordCmd) {
+        PPX_LOG_WARN("Disable ImGui when enable prebuilt command buffer.");
+        mSettings.enableImGui = false;
+    }
+
+    if (mSettings.grfx.numFramesInFlight > mSettings.grfx.swapchain.imageCount) {
+        PPX_LOG_WARN("numFramesInFlight should not exceed image count in swapchain.");
+        mSettings.grfx.numFramesInFlight = mSettings.grfx.swapchain.imageCount;
+    }
+
+    if (mSettings.grfx.numFramesInFlight > 1) {
```

Comment:
I don't think setting the image count to the number of frames in flight is always desired, and it's certainly confusing for the user setting the two options. You can have X frames in flight and Y swapchain images where X != Y and still work (because you block if X < Y, and you have extra images if Y > X).

Reply:
Yes, it would introduce additional complexity. But what I'm saying is, we're exposing these two options to the user and it's weird that we're then forcing them to match in some way. Maybe a better way is to rework the settings: an example is to only have swapchainCount and then a boolean `multipleFramesInFlight` that will automatically assume we use `num frames in flight = swapchain count`, instead of two integer options.

So you can have
```
 mSettings.grfx.swapchain.imageCount = 3;
 mSettings.grfx.multipleFramesInFlight = true;
```
which will be equivalent to the current `mSettings.grfx.numFramesInFlight  = 3`
or 
```
 mSettings.grfx.swapchain.imageCount = 3;
 mSettings.grfx.multipleFramesInFlight = false;
```

which will be equivalent to the current `mSettings.grfx.numFramesInFlight = 1`.